events_backend/app/views.py

```diff
@@ -683,15 +692,100 @@ def get(self, request, mobile_id, format=None):
             return HttpResponse("Reset Token Error")
 
 
-class UploadImage(APIView):
-    permission_classes = (permissions.IsAuthenticated,)
+def upload_image(s3, bucket_name, filename_path, file_data):
+
+    try:
+        print("\nattempting to upload:")
+        print(filename_path)
+        s3.put_object(Bucket=bucket_name, Key=filename_path, Body=file_data.file)
+    except ClientError as e:
+        logging.error(e)
+        return False, ""
+    finally:
+        return True, filename_path
+
+def make_public(s3, bucket_name, filename_path):
+    try:
+        s3.put_object_acl(ACL="public-read",
+        Bucket=bucket_name)
+    except ClientError as e:
+        logging.error(e)
+        return False
+    finally:
+        return True
+
+class UploadImageS3(APIView):
+    authentication_classes = ()
+    permission_classes = ()
 
     def post(self, request):
-        return JsonResponse({
-            "potato": "123"
-        }, status=status.HTTP_200_OK)
+        user_id = str(request.user.id)
+        file_type = request.GET.get("file_type").replace('"', '')
+        uploaded_file_name = request.GET.get("file_name").replace('"', '')
+        
+        fileData = request.data["file"]
+
+        print("reuest data")
+        print(list(request.POST.items()))
+
+
+        S3_BUCKET = settings.AWS_STORAGE_BUCKET_NAME
+        timeString = dt.now().strftime("%Y%m%d_%H%M%S")
+
+
+        s3 = boto3.client(
+            "s3",
+            aws_access_key_id=settings.AWS_ACCESS_KEY_ID,
+            aws_secret_access_key=settings.AWS_SECRET_ACCESS_KEY
+        )
```

Comment:
can probably be a singleton

Reply:
That is definitely a good idea. But just curious, would concurrency affect the usage of the S3 client (race conditions)?